src/autogenerated/java/net/minestom/server/sound/BlockSoundType.java

```diff
@@ -0,0 +1,261 @@
+package net.minestom.server.sound;
+
+import java.util.HashMap;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * AUTOGENERATED by BlockSoundTypesGenerator
+ */
+public record BlockSoundType(float volume, float pitch, @NotNull SoundEvent breakSound,
```

Comment:
This should mimic the other generated resources:
* BlockSoundType interface inheriting from BlockSoundTypes generated with all constants.
* (package private) BlockSoundTypeImpl record containing the relevant fields.

(I think a registry could also work, but that seems to me like it needs a little more discussion. Either way I dont think we should introduce a new style of codegen for it)

Reply:
Done
As said above, I'm not sure it would work in a registry since BlockSoundTypes aren't sent through the network and don't extend StaticProtocolObject.